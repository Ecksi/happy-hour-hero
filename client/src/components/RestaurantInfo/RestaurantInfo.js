import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import './RestaurantInfo.css';
import borderImg from './assets/main-img-border.png';
import moment from 'moment';
moment().format();

export class RestaurantInfo extends Component {  
  render() {
    const { restaurant, joinedTimes, bestDrinkSpecial, bestFoodSpecial } = this.props;
    const { name, address, restaurant_image } = restaurant;
    const backgroundImage = {backgroundImage: "url(" + restaurant_image + ")"};
    const borderBackground = {backgroundImage: "url(" + borderImg + ")"};

    return (
      <section className="restaurantInfoContainer" style={ backgroundImage }>
        <div className="restaurantImageBorder" style={ borderBackground }></div>
        <section className="restaurantInfo">
          <h1>{ name }</h1>
          <p className="restaurantAddress">{ address }</p>
          <h3>happy hour times</h3>
          <p className="times">{ joinedTimes }</p>
          <div className="bestDeals">
            <p><em>best deal</em></p>
            { bestDrinkSpecial ? <p> {bestDrinkSpecial} </p> : null }
            { bestFoodSpecial ? <p> {bestFoodSpecial} </p> : null }
          </div>
        </section>
      </section>
    );
  }
}

RestaurantInfo.propTypes = {
  foodSpecials: PropTypes.array,
  drinkSpecials: PropTypes.array,
  getRestaurant: PropTypes.func,
  getTodaysHappyHours: PropTypes.func,
  cleanHappyHourTimes: PropTypes.func,
  restaurant: PropTypes.object,
  joinedTimes: PropTypes.string,
  bestDrinkSpecial: PropTypes.string,
  bestFoodSpecial: PropTypes.string,
};

export const mapStateToProps = (state) => ({
  restaurantId: state.restaurantId,
  filteredRestaurants: state.filteredRestaurants,
  happyHours: state.happyHours,
  day: state.day,
  drinkSpecials: state.drinkSpecials,
  foodSpecials: state.foodSpecials
});

export default connect(mapStateToProps)(RestaurantInfo);
