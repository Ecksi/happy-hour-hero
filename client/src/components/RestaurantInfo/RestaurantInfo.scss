.restaurantInfoContainer {
  height: 375px;
  background-size: cover;
  background-position-y: center;
}

.restaurantInfo {
  width: calc(80% - 100px);
  padding: 50px;
  left: 10%;
  top: 200px;
  bottom: 100px;
  position: absolute;
  height: 175px;
  background: #fff;
  border-radius: 20px;
  box-shadow: 1px 1px 1px 1px #cccc;
}

.restaurantInfo h1 {
  color: #666;
  margin: 0;
  font-size: 2rem;
  line-height: 2rem;
}

.restaurantInfo h3 {
  color: #666;
  font-size: 1.1rem;
  margin: 80px 0 2px 0;
  font-weight: 400;
}

.restaurantInfo .times {
  color: #b3b3b3;
  font-size: 1.1rem;
  margin-top: 0;
}

.restaurantImageBorder {
  height: 130px;
  position: absolute;
  top: 350px;
  width: 100%;
}

.restaurantAddress {
  color: #b3b3b3;
  font-size: 1.2rem;
  font-weight: 100;
  margin-top: 5px;
}

.bestDeals {
  background: #fdb82a;
  padding: 25px 60px 25px 40px;
  position: absolute;
  right: 0;
  top: 40px;
  border-radius: 20px 0 0 20px;
  color: #fff;
  font-size: 1.3rem;
}

.bestDeals p {
  margin: 0;
}

.bestDeals em {
  color: #ccc;
}

.bestDealTitle {
  font-weight: 100;
  margin-bottom: 0;
}

@media only screen and (max-width: 700px) {
  .restaurantInfo {
    display: flex;
    justify-content: center;
    flex-direction: column;
    align-items: center;
    padding: 50px 50px 100px 50px;
  }

  .restaurantInfo h1 {
    display: block;
    justify-content: center;
    text-align: center;
  }

  .restaurantInfo h3 {
    margin: 0px 0 5px 0;
  }

  .bestDeals {
    position: relative;
    font-size: 1rem;
    border-radius: 20px;
    text-align: center;
    padding: 15px 35px;
  }

  .restaurantAddress {
    margin-top: 5px;
  }

  .restaurantInfo .times {
    margin-bottom: -15px;
  }

  .restaurantHappyHoursContainer {
    text-align: center;
    padding-top: 20px;
  }

  .restaurantSpecialsContainer h2 {
    text-align: center;
    font-size: 1.6rem !important;
  }

  .restaurantHappyHoursContainer h3 {
    font-size: 1.3rem  !important;
  }

  .restaurantHappyHoursContainer h4 {
    font-size: 1.3rem !important;
  }

  .contactBarContainer a {
    font-size: 0.8rem !important;
    flex-direction: column-reverse;
    justify-content: center !important;
    width: 24% !important;

  }

  .contactBarContainer {
    width: calc(80% - 40px) !important;
    padding: 20px !important;
  }

  .contactBarContainer i {
    margin: 0 0 5px 0 !important;
  }
}
