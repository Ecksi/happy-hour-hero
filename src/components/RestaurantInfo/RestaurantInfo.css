.restaurantInfoContainer {
  height: 375px;
  background-size: cover;
  background-position-y: center; }

.restaurantInfo {
  width: calc(80% - 100px);
  padding: 50px;
  left: 10%;
  top: 200px;
  bottom: 100px;
  position: absolute;
  height: 175px;
  background: #fff;
  border-radius: 20px;
  box-shadow: 0px 2px 1px 1px #eee; }

.restaurantInfo h1 {
  color: #666;
  margin: 0;
  font-size: 2rem;
  line-height: 2rem; }

.restaurantInfo h3 {
  color: #666;
  font-size: 1.1rem;
  margin: 80px 0 2px 0;
  font-weight: 400; }

.restaurantInfo .times {
  color: #b3b3b3;
  font-size: 1.1rem;
  margin-top: 0; }

.restaurantImageBorder {
  height: 130px;
  position: absolute;
  top: 350px;
  width: 100%; }

.restaurantAddress {
  color: #b3b3b3;
  font-size: 1.2rem;
  font-weight: 100;
  margin-top: 5px; }

.bestDeals {
  background: #fdb82a;
  padding: 25px 45px;
  position: absolute;
  right: 0;
  top: 40px;
  color: #fff;
  font-size: 1.3rem; }

.bestDeals p {
  margin: 0; }

.bestDealTitle {
  font-weight: 100;
  margin-bottom: 0; }
