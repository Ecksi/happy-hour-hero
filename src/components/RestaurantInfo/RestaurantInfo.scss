.restaurantInfoContainer {
  height: 350px;
  background-size: cover;
  background-position-y: center;
}

.restaurantInfo {
  width: 80%;
  left: 10%;
  top: 200px;
  bottom: 100px;
  position: absolute;
  height: 250px;
  background: #fff;
  border-radius: 20px;
  box-shadow: 2px 2px 2px 2px #eee;
}

.restaurantImageBorder {
  height: 130px;
  position: absolute;
  top: 300px;
  width: 100%;
}
