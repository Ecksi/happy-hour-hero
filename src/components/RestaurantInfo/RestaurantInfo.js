import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import './RestaurantInfo.css';
import borderImg from './assets/main-img-border.png';
import moment from 'moment';
moment().format();

class RestaurantInfo extends Component {  
  render() {
    const { restaurant , todaysHappyHours, joinedTimes, bestDrinkSpecial, bestFoodSpecial } = this.props;

    const { name, address, restaurant_image } = restaurant;
    const backgroundImage = {backgroundImage: "url(" + restaurant_image + ")"};
    const borderBackground = {backgroundImage: "url(" + borderImg + ")"};

    return (
      <section className="restaurantInfoContainer" style={ backgroundImage }>
        <div className="restaurantImageBorder" style={ borderBackground }></div>
        <section className="restaurantInfo">
          <h1>{ name }</h1>
          <p className="restaurantAddress">{ address }</p>
          <h3>happy hour times</h3>
          <p className="times">{ joinedTimes }</p>
          <div className="bestDeals">
            <p><em>best deal</em></p>
            { bestDrinkSpecial ? <p> {bestDrinkSpecial} </p> : null }
            { bestFoodSpecial ? <p> {bestFoodSpecial} </p> : null }
          </div>
        </section>
      </section>
    );
  }
}

RestaurantInfo.propTypes = {
  foodSpecials: PropTypes.string,
  drinkSpecials: PropTypes.string,
  getRestaurant: PropTypes.func,
  getTodaysHappyHours: PropTypes.func,
  cleanHappyHourTimes: PropTypes.func,
};

export const mapStateToProps = (state) => ({
  restaurantId: state.restaurantId,
  filteredRestaurants: state.filteredRestaurants,
  happyHours: state.happyHours,
  day: state.day,
  drinkSpecials: state.drinkSpecials,
  foodSpecials: state.foodSpecials
});

export default connect(mapStateToProps)(RestaurantInfo);
