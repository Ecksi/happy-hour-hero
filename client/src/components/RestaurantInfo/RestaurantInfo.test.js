import React from 'react';
import { shallow } from 'enzyme';
import { RestaurantInfo, mapStateToProps } from './RestaurantInfo';

describe('RestaurantInfo', () => {
  let wrapper;
  let mockProps;

  beforeEach(() => {
    mockProps = {
      name: 'never',
      address: 'gonna',
      restaurant_image: 'give',
      restaurant: 'you',
      joinedTimes: 'up',
      bestDrinkSpecial: 'neva',
      bestFoodSpecial: 'giv',
    };
    wrapper = shallow(<RestaurantInfo {...mockProps} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('can access restaurantId from the store', () => {
      const restId = mockProps.restaurantId;
      const mockState = { restId };
      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(mockState);
    });

    it('can access filteredRestaurants from the store', () => {
      const filteredRests = mockProps.filteredRestaurants;
      const mockState = { filteredRests };
      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(mockState);
    });

    it('can access happyHours from the store', () => {
      const happyHour = mockProps.happyHours;
      const mockState = { happyHour };
      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(mockState);
    });

    it('can access the day from the store', () => {
      const today = mockProps.day;
      const mockState = { today };
      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(mockState);
    });

    it('can access drinkSpecials from the store', () => {
      const drinkSpecial = mockProps.drinkSpecials;
      const mockState = { drinkSpecial };
      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(mockState);
    });

    it('can access foodSpecials from the store', () => {
      const foodSpecial = mockProps.foodSpecials;
      const mockState = { foodSpecial };
      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(mockState);
    });
  });
});