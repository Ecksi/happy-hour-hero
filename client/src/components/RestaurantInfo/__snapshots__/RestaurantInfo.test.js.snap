// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantInfo should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RestaurantInfo
    address="gonna"
    bestDrinkSpecial="neva"
    bestFoodSpecial="giv"
    joinedTimes="up"
    name="never"
    restaurant="you"
    restaurant_image="give"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="restaurantImageBorder"
          style={
                    Object {
                              "backgroundImage": "url(main-img-border.png)",
                            }
          }
/>,
        <section
          className="restaurantInfo"
>
          <h1 />
          <p
                    className="restaurantAddress"
          />
          <h3>
                    happy hour times
          </h3>
          <p
                    className="times"
          >
                    up
          </p>
          <div
                    className="bestDeals"
          >
                    <p>
                              <em>
                                        best deal
                              </em>
                    </p>
                    <p>
                               
                              neva
                               
                    </p>
                    <p>
                               
                              giv
                               
                    </p>
          </div>
</section>,
      ],
      "className": "restaurantInfoContainer",
      "style": Object {
        "backgroundImage": "url(undefined)",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "restaurantImageBorder",
          "style": Object {
            "backgroundImage": "url(main-img-border.png)",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1 />,
            <p
              className="restaurantAddress"
/>,
            <h3>
              happy hour times
</h3>,
            <p
              className="times"
>
              up
</p>,
            <div
              className="bestDeals"
>
              <p>
                            <em>
                                          best deal
                            </em>
              </p>
              <p>
                             
                            neva
                             
              </p>
              <p>
                             
                            giv
                             
              </p>
</div>,
          ],
          "className": "restaurantInfo",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "restaurantAddress",
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "happy hour times",
            },
            "ref": null,
            "rendered": "happy hour times",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "up",
              "className": "times",
            },
            "ref": null,
            "rendered": "up",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  <em>
                                    best deal
                  </em>
</p>,
                <p>
                   
                  neva
                   
</p>,
                <p>
                   
                  giv
                   
</p>,
              ],
              "className": "bestDeals",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <em>
                    best deal
</em>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "best deal",
                  },
                  "ref": null,
                  "rendered": "best deal",
                  "type": "em",
                },
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    "neva",
                    " ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "neva",
                  " ",
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    "giv",
                    " ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "giv",
                  " ",
                ],
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "section",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="restaurantImageBorder"
            style={
                        Object {
                                    "backgroundImage": "url(main-img-border.png)",
                                  }
            }
/>,
          <section
            className="restaurantInfo"
>
            <h1 />
            <p
                        className="restaurantAddress"
            />
            <h3>
                        happy hour times
            </h3>
            <p
                        className="times"
            >
                        up
            </p>
            <div
                        className="bestDeals"
            >
                        <p>
                                    <em>
                                                best deal
                                    </em>
                        </p>
                        <p>
                                     
                                    neva
                                     
                        </p>
                        <p>
                                     
                                    giv
                                     
                        </p>
            </div>
</section>,
        ],
        "className": "restaurantInfoContainer",
        "style": Object {
          "backgroundImage": "url(undefined)",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "restaurantImageBorder",
            "style": Object {
              "backgroundImage": "url(main-img-border.png)",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1 />,
              <p
                className="restaurantAddress"
/>,
              <h3>
                happy hour times
</h3>,
              <p
                className="times"
>
                up
</p>,
              <div
                className="bestDeals"
>
                <p>
                                <em>
                                                best deal
                                </em>
                </p>
                <p>
                                 
                                neva
                                 
                </p>
                <p>
                                 
                                giv
                                 
                </p>
</div>,
            ],
            "className": "restaurantInfo",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "restaurantAddress",
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "happy hour times",
              },
              "ref": null,
              "rendered": "happy hour times",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "up",
                "className": "times",
              },
              "ref": null,
              "rendered": "up",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    <em>
                                        best deal
                    </em>
</p>,
                  <p>
                     
                    neva
                     
</p>,
                  <p>
                     
                    giv
                     
</p>,
                ],
                "className": "bestDeals",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <em>
                      best deal
</em>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "best deal",
                    },
                    "ref": null,
                    "rendered": "best deal",
                    "type": "em",
                  },
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      "neva",
                      " ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    "neva",
                    " ",
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      "giv",
                      " ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    "giv",
                    " ",
                  ],
                  "type": "p",
                },
              ],
              "type": "div",
            },
          ],
          "type": "section",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
